### PR TITLE
JS: clean up babel plugins

### PR DESCRIPTION
**Pull request by fisker**
Fixes https://github.com/prettier/prettier/issues/8430

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parser-babel.js [hunk 1]
```diff
@@ -16,22 +16,17 @@ function babelOptions({ sourceType, extraPlugins = [] }) {
...
       "doExpressions",
-      "objectRestSpread",
       "classProperties",
       "exportDefaultFrom",
-      "exportNamespaceFrom",
-      "asyncGenerators",
       "functionBind",
       "functionSent",
-      "dynamicImport",
+      // "dynamicImport", // Need ref
```
Comment (fisker) [inline src/language-js/parser-babel.js:27]: Can't find a link says this enabled, but removing it seems no effect, the ast is also right 

```text
          "callee": {
            "type": "Import",
            "start": 0,
            "end": 6,
            "loc": {
              "start": { "line": 1, "column": 0 },
              "end": { "line": 1, "column": 6 }
            }
          },
```

File: tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,1529 @@
...
+// https://github.com/js-choi/proposal-smart-pipelines
+
+value
+|> await #
+|> doubleSay(#, ", ")
+|> capitalize  // This is a function call.
+  |> # + "!"
+  |> new User.Message(#)
+  |> await #
+  |> console.log; // This is a method call.
```
Comment (fisker) [inline tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: Why this pipeline format like this? Bug?

Comment (fisker) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: @sosukesuzuki

Comment (sosukesuzuki) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: Which is the input file for the output?

Comment (fisker) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: Shouldn't it  be  

```js
value
  |> await #
  |> doubleSay(#, ", ")
  |> capitalize  // This is a function call.
  |> # + "!"
  |> new User.Message(#)
  |> await #
  |> console.log; // This is a method call.
```

All aligned?

Comment (fisker) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: Seems happens in  `&&` too, I'll open an issue later.

Comment (sosukesuzuki) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: I see. Thank you.

Comment (fisker) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: #8450

Comment (sosukesuzuki) [reply tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1252]: probably you mistook the issue number:smile:

**Pull request merged**